### PR TITLE
Docs/term sequence

### PR DESCRIPTION
**Pull request by CalMacCQ**
Updated the docstring for `gen_term_sequence_circuit`. 

Hopefully it'll be a bit clearer what this method does now. 

Maybe this sort of method would be better to have in an application library going forward.

### CODE REVIEW

File: pytket/pytket/utils/term_sequence.py [hunk 1]
```diff
@@ -28,8 +28,20 @@ def gen_term_sequence_circuit(
...
+
+    The sequencing is done according so some specified partitioning strategy and
+    graph colouring method.
+
+    The resulting Circuit will contain a sequence of CircBoxes. Each CircBox
+    corresponds to a set of Pauli strings. Each exponentiated Pauli string
+    in the set is realised as a PauliExpBox.
+
+    The ordering of terms prioritises reducing the two qubit gate count of the
+    circuit rather than minimising trotter error.
```
Comment (willsimmons1465) [inline pytket/pytket/utils/term_sequence.py:44]: Reduces two qubit gate count _when considering applying PauliSimp or GuidedPauliSimp_. It need not minimise them for other compilation strategies.

**Review by willsimmons1465 — CHANGES_REQUESTED** [review]
A couple of minor adjustments needed. Much better than the original though

File: pytket/pytket/utils/term_sequence.py [hunk 2]
```diff
@@ -28,8 +28,20 @@ def gen_term_sequence_circuit(
     partition_strat: PauliPartitionStrat = PauliPartitionStrat.CommutingSets,
     colour_method: GraphColourMethod = GraphColourMethod.Lazy,
 ) -> Circuit:
-    """Sequences QubitPauliOperator terms to generate a circuit made of CircBoxes.
-    Each CircBox contains a sequence of PauliExpBox objects.
+    """
+    Sequences a the terms of a QubitPauliOperator P in order to generate a circuit
+    approximating :math:`e^{\\frac{\\pi}{2}\\theta P}`. This method performs
```
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/pytket/utils/term_sequence.py:33]: Ah yeah I forgot the imaginary unit, thanks

File: pytket/pytket/utils/term_sequence.py [hunk 3]
```diff
@@ -28,8 +28,20 @@ def gen_term_sequence_circuit(
...
+    Trotterisation with a single Trotter step.
+
+    The sequencing is done according so some specified partitioning strategy and
+    graph colouring method.
```
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/pytket/utils/term_sequence.py]: Thanks, let me now if the revised wording is okay

[the remark this answers is not in the document]
Comment (willsimmons1465) [reply pytket/pytket/utils/term_sequence.py]: Sorry, I meant in the actual API for PauliPartitionStrat: https://github.com/CQCL/tket/blob/366a0a6386112c416f11bf3493840570eeb6692f/pytket/binders/partition.cpp#L33

[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/pytket/utils/term_sequence.py]: Oh I see update the docstring for PauliPartitionStrat. Will do.

File: pytket/pytket/utils/term_sequence.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (willsimmons1465) [reply pytket/pytket/utils/term_sequence.py]: The updated version still contains the unexplained theta

**Review by willsimmons1465 — APPROVED** [review]
Looking good, thanks for fixing this up

**Pull request merged**